### ISSUE TITLE
Persist sent sample sheets

### ISSUE THREAD

**Issue by GerberI** [SiLeBAT/mibi-portal-client#445]
As a user i would like to review my sample sheets sent via the MiBi-Portal. I would like to see the sent sample sheets as a list on a separate _"Order?"_ page. By clicking on a sheet i would like to see the sample data of that sheet in a read only table on a _separate?_ _"Review?"_ page.

_"Order?"_ Page:
As a user i would like to see the sent sample sheets in a list with at least the following information: Date and time the sheet was sent, file name of the sheet, unique id of the sheet.

_"Review?"_ Page:
As a user i would like to review the sample data of an already sent sample sheet. The data should be presented in the same table-form as seen in the "Probendaten" page. The table should be read only. Fields i changed during the send-process should be marked and information of changed data should be shown in a tool-tip by hovering over the field. _Also meta-data i provided during send like analysis procedures should be a accessible somehow (tool-tip, additional columns or pop-up?)._ Also i would like to export the sheet as an excel file.

Common decisions:
- [ ] How should we name the new things? (en+de)
- [x] Should the persisted sheet be splitted by NRL? => No (samples could be assigned to other NRLs outside the portal)
- [ ] Should the persisted sheet contain meta-data => Yes
  - [ ] What meta data should be persisted exactly?
  - [x] Should the meta-data be viewable by the user? => Yes
- [x] Should a unique id be generated for each uploaded sample sheet? => Yes
  - [ ] Should the uid be added to the sent sample sheets?
  - [ ] Should the uid be added to the exported sample sheet?
  - [ ] Should the uid be added to the LIMS?
  - [x] Should the uid be visible by the user => Yes (e.g. to let the support identify a specific sheet)

Server decisions:
- [x] Encrypt sample sheets? Who is allowed to decrypt them? => Yes, first step: decryptable with one master key
- [ ] Should some features be implemented via the parse-server?
  - [ ] db-access?
  - [ ] authorisation?
  - [ ] authentication?
  - [ ] db-encryption?
  - [ ] API?

Client decisions:
- [x] Should the user be able to delete sheets? => No, perhaps in a later step
- Opened sheet
  - [x] Show changed fields (with detailed information) => Yes
  - [x] Should the opened sheet contain validation hints? => No, but perhaps interesting for the admin
  - [x] Should the opened sheet be sendable again? => No, perhaps separate edit sent data feature
  - [x] Should the sheet be openable in a separate read-only view? => Yes
    - [ ] Should the view be shown in the same tab as the orders, in a close-able new tab or a persistent tab?
  - [ ] How should the meta data be presented to the user?

Possible tickets:
- Preparation
  - Data format for user => prepare for multiple users per sample sheet
  - Data format for persisted sample sheet
  - API endpoints
  - Data format for API endpoints
  - Design domain objects
- Server
  - Create domain objects for sample sheet
  - Implement additional data to user entity
  - Implement read/store sample sheet to DB (repositories)
  - Integrate sample sheet persistence in send service (use case)
  - Adapt send API-endpoint
  - Implement user authorization mechanism
  - Create read list use case
  - Create read list endpoint
  - Create read sheet use case
  - Create read sheet endpoint
- Client
  - Implement sheet data objects
  - Implement order page
  - Implement persisted sheet view
  - Implement store connection
  - Implement effects
  - Adapt user data
  - Implement API connection

Mockup Auftragsübersicht :)

* 15.2.2004 sheetabc ergebnisse 5 [open] [export] [delete]

* 15.2.2004 sheetabc ergebnisse 5 [export] [delete]

* 15.2.2004 sheetabc ergebnisse 5 [export] [delete]

* 15.2.2004 sheetabc ergebnisse 5 [export] [delete]

**Comment by dptolle** [comment]
@GerberI Please use the provided pictures as a guide for the required functionality, not as a mock-up for the final design.  MiBi-Portal should adhere as much as possible to the Material Design.  Use Existing Angular Material components where possible (e.g. Table, Paginator etc)

**Comment by GerberI** [comment]
@dptolle Yes, i completely agree. I discussed this with Alex already. The mock-ups are a rough overview meant as a basis for discussion. The final design is a WIP.

**Comment by afalenski** [comment]
Most of this has been done in another way in another issue. Outdated.
